packages/vscode-extension/src/project/DeviceSessionsManager.ts

```diff
@@ -1,298 +1,163 @@
-import { isEqual } from "lodash";
 import { Disposable, window } from "vscode";
-import { BuildError } from "../builders/BuildManager";
-import { DeviceInfo } from "../common/DeviceManager";
-import { ProjectState, StartupMessage } from "../common/Project";
+import { DeviceInfo, DevicePlatform } from "../common/DeviceManager";
 import { DeviceAlreadyUsedError, DeviceManager } from "../devices/DeviceManager";
 import { Logger } from "../Logger";
 import { extensionContext } from "../utilities/extensionContext";
 import { ApplicationContext } from "./ApplicationContext";
-import { DeviceBootError, DeviceSession } from "./deviceSession";
+import { DeviceSession } from "./deviceSession";
 import { AndroidEmulatorDevice } from "../devices/AndroidEmulatorDevice";
 import { IosSimulatorDevice } from "../devices/IosSimulatorDevice";
-import { CancelError } from "../builders/cancelToken";
 import {
   DeviceSessionsManagerInterface,
   DeviceSessionsManagerDelegate,
   ReloadAction,
   SelectDeviceOptions,
 } from "../common/DeviceSessionsManager";
 import { disposeAll } from "../utilities/disposables";
+import { DeviceSessionInitialState } from "../common/Project";
 
 const LAST_SELECTED_DEVICE_KEY = "last_selected_device";
 
-export class DeviceSessionsManager implements DeviceSessionsManagerInterface, Disposable {
-  // selected device id
-  private selectedDevice: string | undefined;
-  private deviceSessions: Map<string, DeviceSession> = new Map();
+export class DeviceSessionsManager implements Disposable, DeviceSessionsManagerInterface {
+  private deviceSessions: Set<DeviceSession> = new Set();
```

Comment:
I'd change this to a map `DeviceId` -> `DeviceSession`, since we probably only want to either
- lookup by device id
- walk over all sessions
But that's not very important when there are probably < 3 sessions at any time.